packages/public-api/src/packageInfo/packageSummaryService.ts

```diff
@@ -0,0 +1,91 @@
+import { getRepository, createQueryBuilder } from 'typeorm';
+import {
+  PackageMetadata,
+  PackagePopularityView,
+  PackageVulnerability,
+  toSerializable,
+} from '@gradejs-public/shared';
+import semver from 'semver';
+
+const simpleVersionRegex = /^\d+\.\d+\.\d+$/;
+
+export async function getPackageSummaryByName(packageName: string) {
+  const packageRepo = getRepository(PackageMetadata);
+  const packagePopularityRepo = getRepository(PackagePopularityView);
+  const packageVulnerabilities = getRepository(PackageVulnerability);
+
+  const queries: [
+    Promise<PackageMetadata | undefined>,
+    Promise<{ hostnamePackagesCount: number; hostname: string; packageName: string }[]>,
+    Promise<number>,
+    Promise<PackagePopularityView | undefined>,
+    Promise<PackageVulnerability[]>
+  ] = [
+    packageRepo.createQueryBuilder().where('name = :packageName', { packageName }).getOne(),
+    createQueryBuilder()
+      .select()
+      .from('package_usage_by_hostname_projection', 'usage')
+      .leftJoin('usage.hostname', 'hn')
+      .addSelect('package_name', 'packageName')
+      .addSelect('hn.hostname', 'hostname')
+      .leftJoin('usage.sourceScan', 'sourceScan')
+      .addSelect(
+        'jsonb_array_length(("sourceScan"."scan_result"->>\'identifiedPackages\')::jsonb)',
+        'hostnamePackagesCount'
+      )
+      .where('package_name = :packageName', { packageName })
+      .limit(16) // TODO: remove hardcode
+      .getRawMany(),
+    createQueryBuilder()
+      .select()
+      .from('package_usage_by_hostname_projection', 'usage')
+      .where('package_name = :packageName', { packageName })
+      .getCount(),
```

Comment:
Counting rows is a relatively slow operation in postgres, we already have a pre-computed field for this purpose: https://github.com/gradejs/gradejs/blob/5991aa6b1b08b6af52e35e2aad225f3f7b8d6bbc/packages/shared/src/database/entities/packagePopularityView.ts#L11